src/main/java/com/documaster/rms/noark/ws/serialization/NoarkOptionalEnumJsonSerializer.java

```diff
@@ -0,0 +1,20 @@
+package com.documaster.rms.noark.ws.serialization;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import com.documaster.rms.noark.ws.constants.NoarkEnum;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+
+public class NoarkOptionalEnumJsonSerializer extends JsonSerializer<Optional<NoarkEnum>> {
+
+	@Override
+	public void serialize(
+			Optional<NoarkEnum> noarkEnum, JsonGenerator jsonGenerator, SerializerProvider serializerProvider)
+			throws IOException {
+
+		jsonGenerator.writeObject(noarkEnum.isPresent() ? noarkEnum.get().getCode() : Optional.empty());
```

Comment:
From what I get we don't reach this code when the `noarkEnum` is null due to the configuration to not serialize nulls, right?

Reply:
Yes, the configuration for ignoring nulls will filter out the null optional fields before reaching the serializer. The same has been relied on in the `NoarkEnumJsonSerializer` from the beginning.